storage/innobase/lock/lock0lock.cc

```diff
@@ -760,6 +760,11 @@ lock_rec_has_to_wait(
 
 		return false;
 	}
+#ifdef WITH_WSREP
+	if (trx->wsrep_UK_scan && wsrep_thd_is_BF(lock2->trx->mysql_thd, true)) {
```

Comment:
Here in my understanding if (trx->wsrep_UK_scan) is enough

Reply:
trx->wsrep_UK_scan tells only that the lock requestor has BF rights and is performing UK scanning atm, but the blocking lock holder should have BF status as well. We want to make the UK GAP exception only between two replayers or appliers.